config/notifiers.go

```diff
@@ -535,6 +535,9 @@ type WebhookConfig struct {
 	// Alerts exceeding this threshold will be truncated. Setting this to 0
 	// allows an unlimited number of alerts.
 	MaxAlerts uint64 `yaml:"max_alerts" json:"max_alerts"`
+
+	// Timeout is the maximum time allowed to invoke the webhook.
+	Timeout *time.Duration `yaml:"timeout" json:"timeout"`
```

Comment:
I'm thinking change this from `*time.Duration` to just `time.Duration`. In this case, a zero value will mean infinite timeout. We do the same for `MaxAlerts` on Line 535 where instead of using a `*uint64` we use the zero value to mean no limit.